src/main/java/lotto/domain/LottoGame.java

```diff
@@ -0,0 +1,60 @@
+package lotto.domain;
+
+import static lotto.domain.model.LottoBall.LOWER_BOUND;
+import static lotto.domain.model.LottoBall.UPPER_BOUND;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import lotto.domain.model.LottoBall;
+
+public class LottoGame {
+
+  public static final int LOTTO_NO_COUNT = 6;
+  private static final int START_INDEX = 0;
+
+  private Set<LottoBall> lottoBalls;
+
+  public LottoGame() {
+    List<Integer> candidates = IntStream.rangeClosed(LOWER_BOUND, UPPER_BOUND)
+        .boxed()
+        .collect(Collectors.toList());
+
+    Collections.shuffle(candidates);
+
+    lottoBalls = candidates.subList(START_INDEX, LOTTO_NO_COUNT)
+        .stream()
+        .map(LottoBall::new)
+        .collect(Collectors.toSet());
+  }
+
+  public LottoGame(Set<Integer> lottoNumbers) {
+    throwIfInvalid(lottoNumbers);
+
+    lottoBalls = lottoNumbers.stream()
+        .map(LottoBall::new)
+        .collect(Collectors.toSet());
+  }
+
+  private void throwIfInvalid(Set<Integer> lottoNumbers) {
+    if (lottoNumbers.size() != LOTTO_NO_COUNT) {
+      throw new IllegalArgumentException("LottoGame should have exactly 6 LottoNos");
+    }
+  }
+
+  public Set<LottoBall> getLottoBalls() {
+    return lottoBalls;
```

Comment:
```suggestion
    return Collections.unmodifiableSet(lottoBalls);
```

Collections.unmodifiableSet()를 활용하여 read-only로 리턴하여 외부에서 값의 변조를 막을 수 있습니다.

https://www.geeksforgeeks.org/collections-unmodifiablelist-method-in-java-with-examples/
https://stackoverflow.com/questions/51881801/why-is-it-better-to-return-unmodifiablelist-from-a-method-where-list-is-created

다만, 이 경우에도 Set에 add, remove 등을 제한할 뿐 속성을 변화시키는 것을 제한하지는 않습니다.
깊은복사를 위해서는 불변객체를 활용해야 합니다. Collection을 감싼 wrapper class를 불변객체로 구성할 것인지는 성능과 고려하여 상황에 따라 판단하는 것이 좋다고 생각합니다.

> https://os94.tistory.com/153

Reply:
unmodifiableSet을 이용하면, collection 자체는 같은 객체를 가리키지만 수정 관련한 method가 지원되지 않는군요.
LottoBall이 VO이기 때문에 unmodifiableSet을 이용하면 원본값이 변화로부터 안전할 것 같네요!
피드백 감사합니다 :)